Python/ceval.c

```diff
@@ -470,6 +471,16 @@ _PyEval_MatchClass(PyThreadState *tstate, PyObject *subject, PyObject *type,
     if (PyObject_IsInstance(subject, type) <= 0) {
         return NULL;
     }
+    // Subpatterns are not supported for union types:
+    if (_PyUnion_Check(type)) {
```

Comment:
I don't like the fact that we call `_PyUnion_Check(type)` twice for no real reason.

Reply:
Could we save the result in a variable? Or would it cleaner to handle unions first?